packages/nuxt/src/app/composables/route-announcer.ts

```diff
@@ -0,0 +1,102 @@
+import type { Ref } from 'vue'
+import { getCurrentScope, onScopeDispose, ref } from 'vue'
+import { useNuxtApp } from '#app'
+
+export enum Politeness {
+  Assertive = 'assertive',
+  Polite = 'polite',
+  Off = 'off',
+}
+
+export type PolitenessValue = `${Politeness}`;
+
+export type NuxtRouteAnnouncerOpts = {
+  /** @default 'polite' */
+  politeness?: PolitenessValue
+}
+
+export type RouteAnnouncer = {
+  message: Ref<string>
+  politeness: Ref<PolitenessValue>
+  set: (message: string, politeness: PolitenessValue) => void
+  polite: (message: string) => void
+  assertive: (message: string) => void
+  _cleanup: () => void
+}
+
+function createRouteAnnouncer (opts: NuxtRouteAnnouncerOpts = {}) {
+  const message = ref('')
+  const politeness = ref<PolitenessValue>(opts.politeness || Politeness.Polite)
+  const nuxtApp = useNuxtApp()
+  let rafId: number | null = null
+  let unsubscribeLoadingFinishHook: () => void = () => {}
+  let _cleanup: () => void = () => {}
+
+  function set (messageValue: string = '', politenessSetting: PolitenessValue = Politeness.Polite) {
+    message.value = messageValue
+    politeness.value = politenessSetting
+  }
+
+  function polite (message: string) {
+    return set(message, Politeness.Polite)
+  }
+
+  function assertive (message: string) {
+    return set(message, Politeness.Assertive)
+  }
+
+  function _updateMessageWithPageHeading () {
+    set(document?.title?.trim(), politeness.value)
```

Comment:
this might be tricky as the title might not be updated synchronously

cc: @harlan-zw - is there an unhead hook we could use, or can you think of a better way to implement?

Reply:
@danielroe thanks for the idea, Unhead hook `dom:rendered` looks much more suitable indeed.

Let see what @harlan-zw will answer, I'll push the commit with refactoring using `dom:rendered` meanwhile

Reply:
✅ Refactored with `dom:rendered` hook